h2o-algos/src/main/java/hex/tree/dt/CompressedDT.java

```diff
@@ -22,13 +21,23 @@ public class CompressedDT extends Keyed<CompressedDT> {
      * For a leaf: (1, decision value, probability), for an internal node: (0, feature index, threshold)
      */
     private final double[][] _nodes;
+    private final AbstractCompressedNode[] _nodesObj;
 
     private final ArrayList<String> _listOfRules;
```

Comment:
List is not allowed as a member of the CompressedDT class.  Use an array instead.  You will have to expand the array size every time it goes out of bound or you can set it to some good estimate of its final size.

This may caused errors in your tests.